docs: add guide on writing mcps to a file (#12654)
datahub-project/datahub: docs/advanced/writing-mcps.md

> # Writing MCPs as Files

Reviewer: I'd prefer to title this "Saving MCPs to a File"

> # Writing MCPs as Files

Author: ```suggestion
# Saving MCPs to a File
```

## What is an MCP?

A `MetadataChangeProposal` (MCP) represents an atomic unit of change in the DataHub Metadata Graph. Each MCP carries a single aspect in its payload and is used to propose changes to DataHub's metadata. 

- Represents a single aspect change
- Used for proposing metadata changes to DataHub
- Serves as the basic building block for metadata ingestion

> For more information, please see [Guide in MCPs](https://datahubproject.io/docs/advanced/mcp-mcl/)

Reviewer: we should also link to this https://datahubproject.io/docs/what/mxe

also, use relative links so that these don't break when we release new docs versions

{
  "entityType": string,          // Type of entity (e.g., "dataset", "chart")
  "entityUrn": string,          // URN of the entity being updated
  "changeType": string,         // Type of change (UPSERT, CREATE, DELETE, etc.)
  "aspectName": string,         // Name of the aspect being modified
  "aspect": {                   // The aspect content
    "value": bytes,            // Serialized aspect value
    "contentType": string      // Serialization format (usually "application/json")
  }
}

```

### Example MCP File

Here's a practical example of an MCP for a dataset:

```json
{
  "entityType": "dataset",
  "entityUrn": "urn:li:dataset:(urn:li:dataPlatform:hive,example_dataset,PROD)",
  "changeType": "UPSERT",
  "aspectName": "datasetProperties",
  "aspect": {
    "value": {
      "description": "Example dataset description",
      "customProperties": {
        "encoding": "utf-8"
      }
    },
    "contentType": "application/json"
  }
}

```

## Why Write MCPs as Files?

MCPs in JSON file format are particularly valuable because they represent the lowest and most granular form of events in DataHub. There are two main use cases for working with MCP files:

### Testing

> MCPs allow you to test your metadata ingestion without setting up complex connector dependencies. You can:

Reviewer: ```suggestion
MCPs allow you to easily ingest metadata. You can:
```

You can export MCPs directly from your DataHub instance using a recipe file. This is useful when you want to:

- Examine existing entities in your DataHub instance
- Create test cases based on real data
- Debug entity relationships

First, create a recipe file (e.g., `export_mcps.yaml`):

```python
source:
  type: datahub
  config:
    # Add your DataHub connection configuration here
    server: "http://localhost:8080"
    token: "your-access-token"  # If authentication is required

sink:
  type: "file"
  config:
    filename: "mcps.json"
```

Run the ingestion:

```python
datahub ingest -c export_mcps.yaml
```

This will write all the entities from your DataHub instance to `mcps.json` in MCP format.

### Creating MCPs with Python SDK

You can use the `FileEmitter` class to generate MCPs programmatically:

```python
import argparse
from datahub.emitter.mcp import MetadataChangeProposalWrapper
from datahub.metadata.schema_classes import (
    DatasetPropertiesClass,
)

import time

from datahub.ingestion.api.common import PipelineContext, RecordEnvelope
from datahub.ingestion.api.sink import NoopWriteCallback
from datahub.ingestion.sink.file import FileSink, FileSinkConfig

> class FileEmitter:

Reviewer: we should recommend the `write_metadata_file` helper method instead of this
